subprojects/docs/src/docs/userguide/workingWithFiles.adoc

```diff
@@ -15,47 +15,288 @@
+    from "${buildDir}/reports/my-report.pdf", "src/docs/manual.pdf"
+    into "${buildDir}/toArchive"
+}
+----
+
+Now two files are copied into the archive directory.
+
+What if you want to copy all the PDFs in a directory without having to specify each one? You can easily attach both inclusion and exclusion patterns to the copy specification. Here we use a pattern to include only PDFs:
+
+----
+task copyReports(type: Copy) {
+    from "${buildDir}/reports"
+    include "*.pdf"
+    into "${buildDir}/toArchive"
+}
+----
+
+One thing to note in this example is that only PDFs directly in the `reports` directory are copied. If there are PDFs in subdirectories of `reports`, they'll be left out. You can change that by using an Ant-style glob pattern:
+
+----
+task copyReports(type: Copy) {
+    from "${buildDir}/reports"
+    include "**/*.pdf"
+    into "${buildDir}/toArchive"
+}
+----
+
+This approach has the side effect of copying the directory structure below `reports` as well as the files. This may be what you want. It may not be. Copying files includes many such decisions and there are too many to cover here, but know that there are very few requirements that can't be handled elegantly by Gradle copy specifications.
+
+Head over to ~TODO~ in order to learn all the different options available for getting the exact behavior you want.
+
+=== Copying directory hierarchies
+
+You may have the need to copy not just files, but the directory structure they reside in as well. This is, in fact, the default behavior when you specify a directory as the `from()` argument. So the following will copy everything in the `reports` directory, including all its subdirectories:
+
+----
+task copyReports(type: Copy) {
+    from "${buildDir}/reports"
+    into "${buildDir}/toArchive"
+}
+----
+
+The key aspect that users struggle with is controlling how much of the directory structure goes to the destination. In the above example, do you get a `toArchive/reports` directory or does everything in `reports` go straight into `toArchive`? The answer is the latter. If a directory is part of the `from()` path, then it *won't* appear in the destination.
+
+So how do you ensure that `reports` itself is copied across, but not any other directory in `$buildDir`? The answer is to add it as an include pattern:
+
+----
+task copyReports(type: Copy) {
+    from("${buildDir}") {
+        include "reports/**"
+    }
+    into "${buildDir}/toArchive"
+}
+----
+
+You'll get the same behavior as before except with one extra level of directory in the destination, i.e. `toArchive/reports`.
+
+One thing to note is how the `include()` directive applies only to the `from()`, whereas the directive in the previous section applied to the whole task. These different levels of granularity in the copy specification allow you to easily handle most requirements that you will come across. TODO: Link to more details on copy specs.
+
+=== Creating archives (zip, tar, etc.)
```

Comment:
Frequently asked about on the user forum: How do I create an uberjar? It would be helpful to explain it here.

Reply:
Good idea. If it's that frequently asked, it might make sense to include it as one of the early examples in the chapter, with perhaps some more detail in the in-depth section.

Reply:
- [ ] Provide an example on creating an uber JAR

@bmuschko Do you mean one in which the contents of the other JARs are included in the fat JAR, or the JARs themselves are included as is in the fat JAR (the Spring Boot approach — or at least it was).